zboxcore/sdk/copyworker.go

```diff
@@ -41,107 +42,167 @@ func (req *CopyRequest) getObjectTreeFromBlobber(blobber *blockchain.StorageNode
 	return getObjectTreeFromBlobber(req.ctx, req.allocationID, req.allocationTx, req.remotefilepath, blobber)
 }
 
-func (req *CopyRequest) copyBlobberObject(blobber *blockchain.StorageNode, blobberIdx int) (fileref.RefEntity, error) {
-	refEntity, err := req.getObjectTreeFromBlobber(req.blobbers[blobberIdx])
+func (req *CopyRequest) copyBlobberObject(
+	blobber *blockchain.StorageNode, blobberIdx int) (refEntity fileref.RefEntity, err error) {
+
+	defer req.wg.Done()
```

Comment:
move the wg.Done outside of this function to where it is defined and avoid shared wg. This function would be misused and cause unexpected issue if the caller didn't init the wg correctly. Let's make it not that easy to go wrong.